pkgs/development/python-modules/einops/default.nix

```diff
@@ -43,8 +43,15 @@ buildPythonPackage rec {
   EINOPS_SKIP_CUPY = 1;
 
   checkPhase = ''
-    export HOME=$TMPDIR
+    # Leftover files in a $TMPDIR can screw tests up, so creating a separate dir
+    export home_backup=$HOME
+    export HOME=$(mktemp -d)
+
     nosetests -v -w tests
+
+    rm -rf $HOME
+    export HOME=$home_backup
+    unset home_backup
```

Comment:
I'm not sure you really have to bother tidying up after yourself or setting `HOME` back to its previous setting. Many packages set `HOME` in the `checkPhase` and I'm not sure I've ever seen any bother setting it back.

Reply:
Also, completely replaced `checkPhase`s could do with `runHook preCheck` and `runHook postCheck` and the beginning and end respectively.

Reply:
I've been wondering about that, why is that needed? Does python add something to those phases, or is it for easier overriding?

Reply:
The hooks? AFAIK It's purely for the convenience of people who may need to override elements of the package and e.g. set their own extra `preCheck`, expecting it to work rather than being ignored.

Reply:
> Many packages set HOME in the checkPhase and I'm not sure I've ever seen any bother setting it back.

It was more of a precaution, so removed it.

> Also, completely replaced checkPhases could do with runHook preCheck and runHook postCheck and the beginning and end respectively.

Added, thanks a lot for a suggestion!